### PR TITLE
Data-plane specs added for Library Apis

### PR DESCRIPTION
**Pull request by navneet35**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/examples/Library_ListByWorkspace.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/12345678-1234-1234-1234-12345678abc/resourceGroups/exampleResourceGroup/providers/Microsoft.Synapse/workspaces/exampleWorkspaceName/libraryartifacts/exampleLibraryName.jar",
+            "name": "exampleLibraryName.jar",
+            "type": "Microsoft.Synapse/workspaces/libraryartifacts",
+            "properties": {
+              "name": "exampleLibraryName.jar",
+              "path": "exampleWorkspace/libraries/exampleLibraryName.jar",
+              "type": "jar",
```
Comment (maha-arun) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/examples/Library_ListByWorkspace.json:24]: add containerName

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/examples/Library_Create.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "parameters": {
+    "endpoint": "exampleWorkspace.dev.azuresynapse.net",
+    "libraryname": "exampleLibraryName.jar",
+    "api-version": "2019-06-01-preview"
+  },
+  "responses": {
+    "202": {
```
Comment (maha-arun) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/examples/Library_Create.json:8]: Should we need add all kinds of failure error codes? If yes, then all of the APIs could return 401, 409, 500 as well. Append, flush could return 404 not found and delete could return 204 no content.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+      "type": "object",
+      "properties": {
+        "content": {
+          "type": "string",
+          "description": "Stream of library content"
+        }
+      }
+    },
+    "OperationResult": {
+      "description": "Operation status for the operation",
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:298]: If this is Azure standard, then we don't need to define its scheme explicitly in this file

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+            "description": "An error response received from the Azure Synapse service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:286]: >definitions [](start = 3, length = 11)

Can we re-use some definitions in the ARM Swagger specs (I see you also send a PR to update ARM APIs: [https://github.com/Azure/azure-rest-api-specs/pull/11418](https://github.com/Azure/azure-rest-api-specs/pull/11418))? This is what we do in the existing data plane Swagger specs, for example: [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#L47](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json#L47) . In this way, we only maintain one piece of object scheme definition so we can decrease the risk of inconsistency between ARM and data plane APIs

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+    "LibraryResource": {
+      "description": "Library response details",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/Library",
```
Comment (idear1203) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:350]: >Library [](start = 33, length = 7)

Can we use another name? (Note this doesn't require any backend API change. It only influence the SDK class name). I got the following error when generate track 2 C# SDK:
Error AZC0012 Single word class names are too generic and have high chance of collision with BCL types or types from other libraries

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/LibraryNameParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted."
+          },
```
Comment (anuchandy) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:118]: same question on whether this is a long running operation, if so mark it as lro.

Comment (anuchandy) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:118]: same question for patch api.

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/libraries/{libraryName}/create": {
+      "put": {
+        "tags": [
+          "Library"
+        ],
+        "operationId": "Library_Create",
```
[the remark this answers is not in the document]
Comment (navneet35) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:67]: 1. Does service support "updating" a library resource, or is it a "create" only resource? - This is a "create" only. 
2. What will happen if someone call this endpoint more than once with same libraryName value? - Then user will get a response that it already exists.

[the remark this answers is not in the document]
Comment (idear1203) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:67]: We haven't gone through the board review but we plan to do sooner in this month

File: specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json [hunk 8]
```diff
@@ -0,0 +1,532 @@
...
+              "format": "binary",
+              "type": "string"
+            }
+          },
+          {
+            "name": "x-ms-blob-condition-appendpos",
+            "in": "header",
+            "description": "Set this header to a byte offset at which the block is expected to be appended. The request succeeds only if the current offset matches this value. Otherwise, the request fails with the AppendPositionConditionNotMet error (HTTP status code 412 – Precondition Failed)",
+            "required": false,
+            "type": "string"
```
Comment (mahariha) [inline specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:260]: long

Comment (navneet35) [reply specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/library.json:260]: will make the change. Thanks!

**Comment by anuchandy** [comment]
@navneet35, please fix the reported ci errors: https://github.com/Azure/azure-rest-api-specs/pull/11423#issuecomment-759474503

**Review by anuchandy — APPROVED** [review]

**Pull request merged**